doc/source/ray-observability/state/state-api.rst

```diff
@@ -0,0 +1,303 @@
+:ref:`Workers (Ray worker processes)`
+:ref:`Runtime environments <runtime-environments>`
+
+E.g. List all nodes
+
+.. code-block:: bash
+
+    ray list nodes 
+
+E.g. List all placement groups
+
+.. code-block:: bash
+
+    ray list placement-groups
+
+Listing resources with one or multiple filters 
+List results could also be filtered. For example: 
+E.g. List local referenced objects created by a process
+
+.. code-block:: bash
+
+    ray list objects -f pid=12345 -f reference_type=LOCAL_REFERENCE
+
+E.g. List alive actors
+
+.. code-block:: bash
+
+    ray list actors -f state=ALIVE
+
+E.g. List running tasks
+
+.. code-block:: bash
+
+    ray list tasks -f scheduling_state=RUNNING
+
+E.g. List non-running tasks
+
+.. code-block:: bash
+
+    ray list tasks -f shceduling_state!=RUNNING
+
+Get
+---
+
+E.g. Get a task info
+
+.. code-block:: bash
+
+    ray get tasks <worker_id> 
+
+E.g. Get a node info
+
+.. code-block:: bash
+
+    ray get nodes <node_id> 
+
+
+Logs
+----
+
+State API also allows you to conveniently access ray logs. Note that you cannot access the logs from a dead node.
+By default, the API prints log from a head node.
+
+E.g. Get all retrievable log file names
+
+.. code-block:: bash
+
+    ray logs 
+
+E.g. Get a particular log file from a node
+
+.. code-block:: bash
+
+    # You could get the node id / node ip from `ray list nodes` 
+    ray logs gcs_server.out --node-id <XYZ> 
+
+E.g. Stream a log file from a node
+
+.. code-block:: bash
+
+    ray logs -f raylet.out --node-ip 172.31.47.143
+
+E.g. Stream actor log with actor id 
+
+.. code-block:: bash
+
+    # You could use ray list actors to get the actor ids
+    ray logs --actor-id=<XXX>
+
+E.g. Stream log from a pid 
+
+.. code-block:: bash
+
+    ray logs --pid=<XXX> --follow
+
+Failure Semantics
+-----------------
+
+The state APIs don't guarantee to return the correct snapshot of the cluster all the time. By default,
+all Python SDKs raise an exception when there's a missing output from the API. And CLI returns a partial result
+and provides warning messages. Here are cases where there can be missing output from the API.
+
+Query Failures
+~~~~~~~~~~~~~~
+
+State APIs query "data sources" (e.g., GCS, raylets, etc.) to obtain and build the snapshot of the cluster.
+However, data sources are sometimes unavailable (e.g., the source is down or overloaded). In this case, APIs
+will return the partial snapshot of the cluster, and users are informed that the output is incorrect through a warning message.
+All warnings are printed through Python's ``warnings`` library, and they can be suppressed.
+
+Data Truncation
+~~~~~~~~~~~~~~~
+
+When the returned data is too large (> 100K), state APIs truncate the output data to ensure system stability.
+Currently, there's no way to control truncated data. In this case, it is recommended to
+reduce the output size using a ``--filter`` option. When truncation happens it will be informed through Python's
```

Comment:
This is actually not true right? Since we do truncation before applying filters. I don't think filters will help.